datanode/src/slippy_util.py

```diff
@@ -0,0 +1,196 @@
+limitations under the License.
+"""
+# logging is our log infrastructure used for this application
+import logging
+import math
+
+# Configure the logger
+logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.INFO)
+log = logging.getLogger('InterUSS_DataNode_SlippyUtil')
+
+
+def ConverCSVtoCoordinates(csv):
+  """Converts and validates string of CSV coords into array of lat/lon coords."""
+  if not csv:
+    raise TypeError('CSV of coordinates must be a string in lat,lon[,lat,lon]...format')
+  result = []
+  coords = csv.split(',')
+  if len(coords) % 2 != 0:
+    raise ValueError('CSV of coordinates must in lat,lon pairs')
+  log.debug('Split coordinates to %s and passed early validation...', coords)
+  for a, b in _Pairwise(coords):
+    lat = float(a)
+    lon = float(b)
+    if lat >= 90 or lat <= -90 or lon >= 180 or lon <= -180:
+      raise ValueError('CSV of coordinates must have valid lat,lon values')
+    result.append((lat, lon))
+  return result
+
+
+def ConvertPointToTile(zoom, latitude, longitude):
+  """Calculation from lat/lon to tile at specific zoom."""
+  if latitude >= 90 or latitude <= -90:
+    raise ValueError('latitude must be within range -90 to 90')
+  if longitude >= 180 or longitude <= -180:
+    raise ValueError('longitude must be within range -180 to 180')
+  if zoom < 0 or zoom > 20:
+    raise ValueError('zoom level must be within range 0 to 20')
+
+  log.debug('ConvertPointToTile for %.3f, %.3f...', latitude, longitude)
+  latitude_rad = math.radians(latitude)
+  n = 2.0 ** zoom
+  xtile = int((longitude + 180.0) / 360.0 * n)
+  ytile = int(
+    (1.0 - math.log(math.tan(latitude_rad) +
+                    (1 / math.cos(latitude_rad))) / math.pi) / 2.0 * n)
+  return xtile, ytile
+
+
+def ConvertPathToTile(zoom, points):
+  """Conversion from a series of lat/lon to tile(s) at the specific zoom.
+
+  This is a bit difficult, since you have to make sure there are enough points
+  in the path to ensure each point touches a tile.
+
+  Args:
+      zoom: level of zoom in slippy terms (0-20)
+      points: array of (lat,lon) that make the path
+  Raises:
+      ValueError if no tiles can be found from the points
+  """
+  log.debug('ConvertPathToTile for %dz and points %s...', zoom, str(points))
+  if not isinstance(points, list):
+    raise TypeError('Points must be a list of (lat,lon)')
+  if len(points) < 2:
+    raise ValueError('Points must be a list of (lat,lon)')
+  if not isinstance(points[0], (tuple,list)):
+    raise TypeError('Points must be a list of (lat,lon)')
+  if len(points[0]) != 2:
+    raise ValueError('Points must be a list of at least two (lat,lon)')
+  last = None
+  result = []
+  for lat, lon in points:
+    if last:
+      latdiff = math.fabs(last[0] - lat)
+      londiff = math.fabs(last[1] - lon)
+      maxdiff = latdiff if latdiff > londiff else londiff
+      splits = int(math.fabs(maxdiff) / DegreesPerTile(zoom)) + 1
```

Comment:
I think there are corner case(s) which might not be covered by this function if the split is 1? 

Example test case: 
```
ConvertPathToTile(9, [(37.936541030367316, -122.377713074509), (37.69672993401783, -122.10422390269278)])
```

The line should cross through three grids (82, 198), (82, 197), and (81,197), but the conversion only returns two (81, 197), and (82, 198) because the delta step is too large. In fact the coordinates above representing a flight between Richmond and Hayward, the predominant portion of the path is in the tile which is not returned (82, 197).  

![image](https://user-images.githubusercontent.com/18078271/46896141-829ed700-ce30-11e8-9bed-e89fb1e0d956.png)

Reply:
Thanks, I hadn't visualized the corner cutters. Will fix.